File: app/src/main/java/rojoxpress/slideexample/MainActivity.java
package rojoxpress.slideexample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.SwitchCompat;
import android.widget.CompoundButton;
import android.widget.TextView;
import android.widget.Toast;

import com.rojoxpress.slidebutton.SlideButton;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final TextView textView = (TextView) findViewById(R.id.progress);
        final SlideButton slideButton = (SlideButton) findViewById(R.id.slide_button);
        SwitchCompat switchCompat = (SwitchCompat) findViewById(R.id.switch_);

        slideButton.setSlideButtonListener(new SlideButton.SlideButtonListener() {
            @Override
            public void onSlide() {
                Toast.makeText(MainActivity.this,"UNLOCKED",Toast.LENGTH_SHORT).show();
            }
        });

        slideButton.setOnSlideChangeListener(new SlideButton.OnSlideChangeListener() {
            @Override
            public void onSlideChange(float position) {
                textView.setText("Progress: "+position);
            }
        });

        switchCompat.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                slideButton.setEnabled(b);
            }
        });
    }
}

File: app/src/main/java/rojoxpress/slideexample/MainActivity.kt
package rojoxpress.slideexample

import android.os.Bundle
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SwitchCompat
import com.rojoxpress.slidebutton.SlideButton
import com.rojoxpress.slidebutton.SlideButton.OnSlideChangeListener
import com.rojoxpress.slidebutton.SlideButton.SlideButtonListener

open class MainActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val textView = findViewById<View>(R.id.progress) as TextView
        val slideButton = findViewById<View>(R.id.slide_button) as SlideButton
        val switchCompat = findViewById<View>(R.id.switch_) as SwitchCompat

        slideButton.setOnSlideListener {
            Toast.makeText(this@MainActivity, "UNLOCKED", Toast.LENGTH_SHORT).show()
        }

        slideButton.setOnSlideChangeListener(object : OnSlideChangeListener {
            override fun onSlideChange(position: Float) {
                textView.text = "Progress: $position"
            }
        })

        switchCompat.setOnCheckedChangeListener { compoundButton, b -> slideButton.isEnabled = b }
    }

}